torch/utils/data/datapipes/iter/combining.py

```diff
@@ -0,0 +1,40 @@
+from torch.utils.data import IterDataPipe, functional_datapipe
+from typing import Iterator, Optional, Sized, Tuple, TypeVar
+
+T_co = TypeVar('T_co', covariant=True)
+
+
+@functional_datapipe('concat')
+class ConcatIterDataPipe(IterDataPipe):
```

Comment:
Do you guys think `ConcatIterDataPipe` is more reasonable considering BC of `ConcatDataSet` or `ChainIterDataPipe` that is similar to python itertools.